x/fees/keeper/integration_test.go

```diff
@@ -0,0 +1,621 @@
+			params.ValidatorShares = sdk.NewDec(0)
+			s.app.FeesKeeper.SetParams(s.ctx, params)
+
+			preFeeColectorBalance := s.app.BankKeeper.GetBalance(s.ctx, feeCollectorAddr, evmDenom)
+			preBalance := s.app.BankKeeper.GetBalance(s.ctx, deployerAddress, evmDenom)
+			gasTipCap := big.NewInt(10000)
+			gasFeeCap := new(big.Int).Add(s.app.FeeMarketKeeper.GetBaseFee(s.ctx), gasTipCap)
+			res := contractInteract(userKey, &contractAddress, nil, gasFeeCap, gasTipCap, &ethtypes.AccessList{})
+
+			developerCoins, _ := calculateFees(params, res, gasFeeCap, 14)
+			feeColectorBalance := s.app.BankKeeper.GetBalance(s.ctx, feeCollectorAddr, evmDenom)
+			balance := s.app.BankKeeper.GetBalance(s.ctx, deployerAddress, evmDenom)
+			Expect(balance).To(Equal(preBalance.Add(developerCoins)))
+			Expect(feeColectorBalance).To(Equal(preFeeColectorBalance))
+			s.Commit()
+		})
+
+		It("we can update the withdrawal address if it is different than the deployer address", func() {
+			params = s.app.FeesKeeper.GetParams(s.ctx)
+			withdrawAddress := sdk.AccAddress(tests.GenerateAddress().Bytes())
+			msg := types.NewMsgUpdateDevFeeInfo(
+				contractAddress2,
+				deployerAddress,
+				withdrawAddress,
+			)
+
+			res := deliverTx(deployerKey, msg)
+			Expect(res.IsOK()).To(
+				Equal(true),
+				"withdraw update failed: "+res.GetLog(),
+			)
+			s.Commit()
+
+			fee, isRegistered := s.app.FeesKeeper.GetFeeInfo(s.ctx, contractAddress2)
+			Expect(isRegistered).To(Equal(true))
+			Expect(fee.ContractAddress).To(Equal(contractAddress2.Hex()))
+			Expect(fee.DeployerAddress).To(Equal(deployerAddress.String()))
+			Expect(fee.WithdrawAddress).To(Equal(withdrawAddress.String()))
+			s.Commit()
+
+			preBalanceD := s.app.BankKeeper.GetBalance(s.ctx, deployerAddress, evmDenom)
+			preBalanceW := s.app.BankKeeper.GetBalance(s.ctx, withdrawAddress, evmDenom)
+			gasPrice := big.NewInt(2000000000)
+			res = contractInteract(userKey, &contractAddress2, gasPrice, nil, nil, nil)
+			s.Commit()
+
+			developerCoins, _ := calculateFees(params, res, gasPrice, 14)
+			balanceD := s.app.BankKeeper.GetBalance(s.ctx, deployerAddress, evmDenom)
+			balanceW := s.app.BankKeeper.GetBalance(s.ctx, withdrawAddress, evmDenom)
+			Expect(balanceW).To(Equal(preBalanceW.Add(developerCoins)))
+			Expect(balanceD).To(Equal(preBalanceD))
+		})
+
+		It("we cannot update the withdrawal address if it is the same as the deployer address", func() {
+			params = s.app.FeesKeeper.GetParams(s.ctx)
+			msg := types.NewMsgUpdateDevFeeInfo(
+				contractAddress2,
+				deployerAddress,
+				deployerAddress,
+			)
+
+			res := deliverTx(deployerKey, msg)
+			Expect(res.IsOK()).To(
+				Equal(false),
+				"withdraw update failed: "+res.GetLog(),
+			)
+			Expect(
+				strings.Contains(res.GetLog(),
+					"withdraw address must be different that deployer address"),
+			).To(BeTrue())
+			s.Commit()
+		})
+
+		It("we can cancel a contract registration, so no fees are distributed to deployer", func() {
```

Comment:
can we add aditional cases:

- updates should fail if there is no contract registered by the deployer
- cancel should fail if there is no contract registered by the deployer

Reply:
Done.